### ISSUE TITLE
engine_http does not throw error if custom js function does not exist

### ISSUE THREAD

**Issue by elton-okawa** [artilleryio/artillery#1099]
Greetings everyone 👋 

I'm using `artillery v1.7.3` and noticed that `engine_http` does not throw an error if custom function does not exist [engine_http (195-209)](https://github.com/artilleryio/artillery/blob/7188c8134a2f8088284b8a8a2309a49e70b02d86/core/lib/engine_http.js#L195-L209)
```
if (requestSpec.function) {
  return function(context, callback) {
    let processFunc = self.config.processor[requestSpec.function];
    if (processFunc) {
      return processFunc(context, ee, function(hookErr) {
        if (hookErr) {
          ee.emit('error', hookErr.code || hookErr.message);
        }
        return callback(hookErr, context);
      });
    } else {
      return process.nextTick(function () { callback(null, context); });
    }
  };
}
```

I think that it'd be better to give an error right there instead of silently ignoring it, that way it'd help us a lot to debug an unusual behavior.

After discussion and approval, I'd like to try to change it.

**Comment by hassy** [comment]
hi @elton-okawa 👋 great suggestion! and thank you for offering your time to implement it too!

the easiest way to implement this would be to use the [metrics API](https://artillery.io/docs/guides/guides/extending.html#Custom-metrics-API) and emit a counter for each of those errors. That would make the issue visible to the end user. Additional information (function name, stack trace etc) can be included in [`debug`](https://github.com/artilleryio/artillery/blob/master/core/lib/engine_http.js#L11) output if needed.

**Comment by elton-okawa** [comment]
Solved by #1102
